### PR TITLE
Allow configuring maxWorkerThreads as a multiplier of processors

### PR DESCRIPTION
**Pull request by phd3**
## Description

Currently configuring maxWorkerThreads is allowed only through a hardcoded value - this enabled configuration relative to processor count.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java [hunk 1]
```diff
@@ -54,7 +54,8 @@
     private DataSize maxLocalExchangeBufferSize = DataSize.of(32, Unit.MEGABYTE);
     private DataSize maxIndexMemoryUsage = DataSize.of(64, Unit.MEGABYTE);
     private boolean shareIndexLoading;
-    private int maxWorkerThreads = Runtime.getRuntime().availableProcessors() * 2;
+    private double maxWorkerThreadsProcessorMultiplier = 2.0;
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java:57]: Maybe instead it should be a String with a custom syntax supporting either: 16 (absolute value), C16 (16 * core count) similiar to Maven parallelism settings

Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java:57]: Or even a custom type: `ThreadCount` that does the validation, parsing etc

Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java:57]: It's good to know the actual number which is printed out during system startup in logs

**Comment by mosabua** [comment]
Is this still in progress @phd3 ? Also is the complexity really worth it and is the processor detection reliable and useful in use cases like on K8s?

**Comment by mosabua** [comment]
According to @phd3 this was replaced by a PR from @wendigo in airlift. As such we can close the PR.

**Pull request closed without merging**